### PR TITLE
Add dry-run option

### PR DESCRIPTION
**Pull request by kpcyrd**
This way you can preview what's going to be posted without actually posting it yet.

### CODE REVIEW

**Comment by kpcyrd** [comment]
I've added another commit that refactors error handling and removes almost all unwraps. Note that I've only lightly tested this.

**Comment by kpcyrd** [comment]
It seems my original implementation is marking tweets as synced, which is basically what I've requested in #9.

I've added another commit that resolves #9 this way and I'm now checking the dry run flag before marking posts as synced.

**Comment by klausi** [comment]
Oh wow, what a super nice cleanup and improvement! Thanks a lot!

The code changes look fine, I will do some manual testing soon and then we can merge this.

File: src/config.rs [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
+use std::fs;
 use std::fs::remove_file;
-use std::fs::File;
-use std::io::prelude::*;
 
-pub fn config_load(mut file: File) -> Config {
-    let mut config = String::new();
-    file.read_to_string(&mut config).unwrap();
-    toml::from_str(&config).unwrap()
+#[inline]
```
Comment (klausi) [inline src/config.rs]: what is the benefit of using inline here? We should add a comment.

**Comment by klausi** [comment]
That was easy, dry-run now also implemented for the delete operations.

So the only very minor remaining thing is to explain the inlining of that function, just let me know!

**Comment by kpcyrd** [comment]
It's a very minor thing, it allows the compiler to replace a call to `let config = config_load("foo.toml");` with the function body of `config_load`, so it would look like `let config = toml::from_str::<Config>(config).map_err(Error::from);`.

It's more of a habbit that I do for very short and trivial functions, we don't really care about performance or binary size here.

If you compile this code:
```
fn foo(x: &str) {
    println!("hello {}", x);
}   

fn bar(x: &str) {
    foo(x)
}   

fn main() {
    bar("world");
}   
```
You can find this in the final output:
```
playground::bar:
	subq	$24, %rsp
	movq	%rdi, 8(%rsp)
	movq	%rsi, 16(%rsp)
	movq	8(%rsp), %rdi
	movq	16(%rsp), %rsi
	callq	playground::foo
	addq	$24, %rsp
	retq

playground::main:
	pushq	%rax
	leaq	.L__unnamed_3(%rip), %rax
	movq	%rax, %rdi
	movl	$5, %esi
	callq	playground::bar
	popq	%rax
	retq
```
If you put `#[inline(always)]` on the bar function the function bar is merged into main (not sure why you have to force it here, but `#[inline]` is optional and lets the compiler decide):
```
playground::main:
	subq	$24, %rsp
	leaq	.L__unnamed_3(%rip), %rax
	movq	%rax, 8(%rsp)
	movq	$5, 16(%rsp)
	movq	8(%rsp), %rdi
	movq	16(%rsp), %rsi
	callq	playground::foo
	addq	$24, %rsp
	retq
```

**Comment by klausi** [comment]
I see, that makes sense. In our case the inlining will not really make a difference and I would leave such micro optimizations for later when performance profiling shows it would be an improvement. Anyway, it does not really matter, so I'll just leave it in.

Binary size: mastodon-twitter-sync is at 14MB :-( . This is really terrible for the simple functionality this implements. And that does not even include the linked libraries like libssl it requires. If only the Rust compiler could only compile the used parts of the 200 dependencies we are using here.

**Pull request merged**